### PR TITLE
Feature: add user common types to pb impl

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1.  add user common types to pb impl.
2. add unit test cases of user pb impl.
3. split old `from_to_protobuf_impl.rs` into `{database|table|data}_rs`.

## Changelog

- New Feature

## Related Issues

Fixes [#issue](https://github.com/datafuselabs/databend/issues/5221)

### CODE REVIEW

File: common/proto-conv/src/lib.rs [hunk 1]
```diff
@@ -18,8 +18,15 @@
...
-mod from_to_protobuf_impl;
+mod table_from_to_protobuf_impl;
+mod user_from_to_protobuf_impl;
+mod util;
 
 pub use from_to_protobuf::FromToProto;
 pub use from_to_protobuf::Incompatible;
+pub use util::check_ver;
+pub use util::missing;
+pub use util::VER;
```
Comment (drmingdrmer) [inline common/proto-conv/src/lib.rs:32]: Nice:D

File: common/protos/proto/user.proto [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    bytes hash_value = 1;
+    PasswordHashMethod hash_method = 2;
+  }
+  message JWT {}
+
+  oneof info {
+    None none = 1;
+    Password password = 2;
+    JWT jwt = 3;
+  }
```
Comment (drmingdrmer) [inline common/protos/proto/user.proto:40]: What aboutt using a single `Empty` message for all kinds of enum variants that do not have a body?
Using `Empty` for `None` and `JWT` will get rid of several empty messages.

I.e., trade specificity for simplicity.

**Review by drmingdrmer — APPROVED** [review]
Great work man!

**Comment by BohuTANG** [comment]
Add: `Part of#5221` to PR summary.
Let's continue the work :rocket:

**Pull request merged**